### PR TITLE
Require Aggregate::rawInputTypes

### PR DESCRIPTION
**Pull request by mbasmanova**
Differential Revision: D50194247

Part of #6506

### CODE REVIEW

**Review by mbasmanova — COMMENTED** [review]
@rui-mo Rui, we are fixing #6506 but requiring AggregationNode to identify aggregate functions not only by name but also by raw input types. This breaks Substrait conversion. I wonder if someone familiar with Substrait could help fix that.

**Comment by rui-mo** [comment]
Hi @mbasmanova, is the Substrait issue solved? We can help on that if needed. cc @PHILO-HE

**Comment by mbasmanova** [comment]
> Hi @mbasmanova, is the Substrait issue solved? We can help on that if needed. cc @PHILO-HE

@rui-mo Hi Rui, I think I figured it out. Thanks for offering to help. It would be nice if someone could take a look at the solution I have.

File: velox/substrait/SubstraitToVeloxPlan.cpp [hunk 1]
```diff
@@ -154,8 +156,29 @@ core::PlanNodePtr SubstraitVeloxPlanConverter::toVeloxPlan(
...
+        rawInputTypes.push_back(
+            VeloxSubstraitSignature::fromSubstraitSignature(typeName));
+        break;
+      }
+
+      std::string typeName = functionSpec.substr(pos + 1, endPos - pos - 1);
+      rawInputTypes.push_back(
+          VeloxSubstraitSignature::fromSubstraitSignature(typeName));
+      pos = endPos + 1;
+    }
```
Comment (rui-mo) [inline velox/substrait/SubstraitToVeloxPlan.cpp:178]: Locally, we implemented this function to extract types from a function signature.
[SubstraitParser.cpp](https://github.com/oap-project/velox/blob/main/velox/substrait/SubstraitParser.cpp#L295-L320)

File: velox/substrait/VeloxSubstraitSignature.cpp [hunk 2]
```diff
@@ -59,6 +59,58 @@ std::string VeloxSubstraitSignature::toSubstraitSignature(const TypePtr& type) {
   }
 }
 
+// static
+TypePtr VeloxSubstraitSignature::fromSubstraitSignature(
```
Comment (rui-mo) [inline velox/substrait/VeloxSubstraitSignature.cpp:63]: Locally we implemented a type mapping from Substrait type name to Velox type name as [SubstraitParser.h](https://github.com/oap-project/velox/blob/main/velox/substrait/SubstraitParser.h#L139-L155) shows, and with this function [TypeUtils.cpp](https://github.com/facebookincubator/velox/blob/main/velox/substrait/TypeUtils.cpp#L68), the string can be converted to Velox type.

**Review by rui-mo — COMMENTED** [review]
Hi @mbasmanova, we added two similar implementations for some of the code in our local repo. Hope that could help here.

File: velox/substrait/VeloxSubstraitSignature.cpp (hunk 2 above)
Comment (mbasmanova) [reply velox/substrait/VeloxSubstraitSignature.cpp:63]: @rui-mo Rui, thank you for the pointers. Would you be interested in contributing these changes?

File: velox/substrait/SubstraitToVeloxPlan.cpp [hunk 3]
```diff
@@ -154,8 +156,29 @@ core::PlanNodePtr SubstraitVeloxPlanConverter::toVeloxPlan(
...
     auto aggExpr = std::make_shared<const core::CallTypedExpr>(
         aggVeloxType, std::move(aggParams), funcName);
+
+    const auto& functionSpec = findFunction(aggFunction.function_reference());
+
+    std::vector<TypePtr> rawInputTypes;
+
+    size_t pos = functionSpec.find(":");
+    for (;;) {
+      size_t endPos = functionSpec.find("_", pos + 1);
```
Comment (xiaoxmeng) [inline velox/substrait/SubstraitToVeloxPlan.cpp:166]:
```
nit: const size_t endPos =
```

File: velox/substrait/SubstraitToVeloxPlan.cpp [hunk 4]
```diff
@@ -154,8 +156,29 @@ core::PlanNodePtr SubstraitVeloxPlanConverter::toVeloxPlan(
...
+      if (endPos == std::string::npos) {
+        std::string typeName = functionSpec.substr(pos + 1);
```
Comment (xiaoxmeng) [inline velox/substrait/SubstraitToVeloxPlan.cpp:168]: ditto

**Review by xiaoxmeng — APPROVED** [review]
@mbasmanova LGTM. Thanks!

File: velox/substrait/VeloxSubstraitSignature.cpp (hunk 2 above)
Comment (rui-mo) [reply velox/substrait/VeloxSubstraitSignature.cpp:63]: Sure. I expect to open the PR next week, and will ping you for review. Thanks.

Comment (rui-mo) [reply velox/substrait/VeloxSubstraitSignature.cpp:63]: Hi @mbasmanova, I opened https://github.com/facebookincubator/velox/pull/7181 for the type conversion part. Would you like to take a review? Thanks.

**Pull request closed without merging**